pkg/ottl/contexts/internal/ottlcommon/attributes_test.go

```diff
@@ -0,0 +1,71 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ottlcommon
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestCheckValuesHomogeneous(t *testing.T) {
+	tcs := []struct {
```

Comment:
Test cases covering empty slices and non-slice values would be nice to have here.

Reply:
What would the expectations be for an empty slice?

Reply:
Empty slices are permitted since pdata slices are untyped.

Reply:
Done